tests/functional/h/hidden_kwarg.py

```diff
@@ -0,0 +1,31 @@
+"""
+The `hidden-kwarg` warning message is emitted when a function is called with
+a keyword argument which shares a name with a positional-only parameter and
+the function contains a keyword variadic parameter dictionary.
+It may be surprising behaviour when the keyword argument is added to the
+keyword variadic parameter dictionary.
+"""
+
+
+def name1(apple, /, banana="Yellow banana", **kwargs):
+    """
+    Positional-only parameter, positional-or-keyword parameter and `**kwargs`.
+
+    `apple` is considered to be a hidden kwarg.
+    `banana` is not hidden since it is a positional-or-keyword parameter.
+
+    >>> name1("Red apple", apple="Green apple", banana="Green banana")
+    Red apple
+    Green banana
+    {"apple": "Green apple"}
+    """
+
+    print(apple)
+    print(banana)
+    print(kwargs)
+
+
+# `apple` is hidden:
+name1("Red apple", apple="Green apple", banana="Green banana")  # [hidden-kwarg]
```

Comment:
Can you add a test case for a signature having a default for `apple=` and only kwargs in the call? I feel like that's the most likely mistake. Such an example would also be good for the `good.py` file.

Reply:
You got it. I don't think I understand the second part of your comment; I think the `good.py` example already has something like you describe; can you clarify please?

Reply:
My fault, I was viewing just a commit and thought I was viewing the PR. Apologies for the rushed comment!

Reply:
No worries! Thanks for your help reviewing 👍